File: Java/WaveSort.java
class WaveSort { 
	static void swap(int arr[], int a, int b) { 
		int temp = arr[a]; 
		arr[a] = arr[b]; 
		arr[b] = temp; 
	} 

	static void sort(int arr[], int n)  {
		for (int i = 0; i < n; i+=2) { 
			if (i>0 && arr[i-1] > arr[i] ) 
				swap(arr, i-1, i); 

			if (i<n-1 && arr[i] < arr[i+1] ) 
				swap(arr, i, i + 1); 
		} 
	} 

	public static void main(String args[]) { 
		int arr[] = {22,10,3,12,21,1,2002,2001}; 
		int n = arr.length; 
		sort(arr, n); 
		
		for (int i : arr) 
			System.out.print(i); 
    
		
	} 
} 

File: CPP/wave-sort.cpp
#include<iostream>

using namespace std;

void swap(int * p, int * q){
  int temp = * p;
  * p = * q;
  * q = temp;
}

void array_in_wave(int array[], int n){
  sort(array, array + n);
  for (int i = 0; i < n - 1; i += 2)
    swap( & array[i], & array[i + 1]);
}

int main(){
  int array[100], n, i;
  cout << "Enter number of elements: ";
  cin >> n;
  cout << "\nEnter elements: ";

  for (i = 0; i < n; i++)
    cin >> array[i];

  cout << "Original array: ";

  for (int i = 0; i < n; i++)
    cout << array[i] << " ";

  array_in_wave(array, n);

  cout << "\nWave form of the array is: ";

  for (int i = 0; i < n; i++)
    cout << array[i] << " ";

  return 0;
}
